File: data/src/main/java/com/marvelcomics/brito/data/repository/series/SeriesRepository.java
package com.marvelcomics.brito.data.repository.series;

import com.marvelcomics.brito.data.datasource.remote.response.SeriesResponse;
import com.marvelcomics.brito.data.datasource.remote.response.model.RemoteMarvelContainer;

import io.reactivex.Observable;

public interface SeriesRepository {

    Observable<RemoteMarvelContainer<SeriesResponse>> series(int characterId);
}

File: data/src/main/java/com/marvelcomics/brito/data/repository/series/SeriesRepository.kt
package com.marvelcomics.brito.data.repository.series

import com.marvelcomics.brito.data.datasource.remote.mapper.SeriesMapper
import com.marvelcomics.brito.data.webservice.MarvelWebService
import com.marvelcomics.brito.domain.entity.SeriesEntity
import com.marvelcomics.brito.domain.repository.ISeriesRepository
import com.marvelcomics.brito.infrastructure.exception.MarvelMapperException

class SeriesRepository(
    private val webService: MarvelWebService,
    private val seriesMapper: SeriesMapper
) : ISeriesRepository {
    override suspend fun getSeries(characterId: Int): List<SeriesEntity> {
        return seriesMapper.transform(webService.series(characterId))
            ?: throw MarvelMapperException("Error mapping series", null)
    }
}
